lib/IR/ChildMemSizeBasedScheduler.cpp

```diff
@@ -149,8 +151,28 @@ void ChildMemSizeBasedScheduler::orderChildNodesAndSchedule(Node *N) {
   }
 
   // Schedule the node after all its children are scheduled.
-  DEBUG_GLOW(llvm::dbgs() << "Scheduled node: " << N->getName() << "\n");
+  if (isScheduled(N)) {
+    return;
+  }
   scheduled_.push_back(N);
+  // If this node has a user which does not have any users and which does not
```

Comment:
I would continue this commit with "because ...".  

... because we don't want to extend the lifetime of this value for no reason. We want to execute and get rid of this node as soon as possible to reduce the memory pressure.